db/python/layers/analysis.py

```diff
@@ -301,17 +232,17 @@ async def add_samples_to_analysis(
                 self.author, project_ids, readonly=False
             )
 
-        return await self.at.add_samples_to_analysis(
-            analysis_id=analysis_id, sample_ids=sample_ids
+        return await self.at.add_sequencing_groups_to_analysis(
+            analysis_id=analysis_id, sequencing_group_ids=sequencing_group_ids
         )
 
     async def update_analysis(
         self,
         analysis_id: int,
         status: AnalysisStatus,
-        meta: Dict[str, Any] = None,
-        output: Optional[str] = None,
-        author: Optional[str] = None,
+        meta: dict[str, Any] = None,
+        output: str | None = None,
+        author: str | None = None,
         check_project_id=True,
     ):
```

Comment:
I think we should come back to this, I'm definitely open to that but at this stage I don't want to touch it, partially because I want to release this with an in-progress hail batch job.